commec/tools/database_handler.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021-2024 International Biosecurity and Biosafety Initiative for Science
+"""
+Abstract base class for a database handler. 
+Customize to screen desired database functionality.
+"""
+import os
+from dataclasses import dataclass
+import subprocess
+import logging
+
+@dataclass
+class DatabaseVersion():
+    """ Container class for outputting version related information from a database."""
+    def __init__(self, input_version : str = "x.x.x", input_date : str = "Null", input_comment :str = ""):
+        self.version_string = input_version
+        self.version_date = input_date
+        self.additional_comment = input_comment
+    version_string : str
+    version_date : str
+    additional_comment : str
+
+class DatabaseHandler():
+    """ 
+    Abstract Class for holding the directory, and file of a database, 
+    including input and output files for screening, as well as screening arguments. 
+    Override the Screen() method for a custom database implementation.
+    """
+    # Start Database Handler API
+    def screen(self):
+        """ Virtual function to be called by any child database implementation"""
+        raise NotImplementedError(
+            "This class must override the .screen() to use DatabaseHandler."
+        )
+
+    def get_version_information(self) -> DatabaseVersion:
+        """ Override to call version retrieval information on a database."""
+        new_version_info = DatabaseVersion("", "", "Version retrieval not yet supported for this database")
+        return new_version_info
```

Comment:
Would it make sense to set the version on `init`, rather than having it as a separate method?

Reply:
Yea why not. We can set version as a member variable.

Reply:
Although will keep the method, override system, as it is neater to call it within super().__init__ that way.